pytorch_pfn_extras/runtime/_runtime.py

```diff
@@ -311,6 +311,17 @@ def map(
         """
         raise NotImplementedError()
 
+    @classmethod
+    @contextlib.contextmanager
+    def trace(cls, event_name: str, arg: Any) -> Generator[None, None, None]:
```

Comment:
This is a class method since we can have multiple runtime instances (one per module created with `ppe.to`). I think we should address these issues in other PR since the ppe.to and nested modules semantics changed and can be tricky.